packages/flutter/lib/src/material/tabs.dart

```diff
@@ -1548,13 +1548,12 @@ class _TabBarState extends State<TabBar> {
 
     final double index = _controller!.index.toDouble();
     final double value = _controller!.animation!.value;
-    final double offset = switch (value - index) {
-      -1.0 || 1.0 => leadingPosition ?? middlePosition,
-      0 => middlePosition,
-      < 0 when leadingPosition == null => middlePosition,
-      > 0 when trailingPosition == null => middlePosition,
-      < 0 => lerpDouble(middlePosition, leadingPosition, index - value)!,
-      _   => lerpDouble(middlePosition, trailingPosition, value - index)!,
+    final double offset = switch ((value, index)) {
+      (_, _) when value == index - 1.0 =>  leadingPosition ?? middlePosition,
+      (_, _) when value == index + 1.0 =>  trailingPosition ?? middlePosition,
+      (_, _) when value == index =>  middlePosition,
+      (_, _) when value < index =>  leadingPosition == null ? middlePosition : lerpDouble(middlePosition, leadingPosition, index - value)!,
+      (_, _) =>  trailingPosition == null ? middlePosition : lerpDouble(middlePosition, trailingPosition, value - index)!,
     };
```

Comment:
It feels a bit weird to pass a record into a switch expression and then not use it.

```suggestion
    final double offset = switch (value - index) {
      -1.0 => leadingPosition ?? middlePosition,
      1.0  => trailingPosition ?? middlePosition,
      0    => middlePosition,
      < 0  => leadingPosition == null ? middlePosition : lerpDouble(middlePosition, leadingPosition, index - value)!,
      _    => trailingPosition == null ? middlePosition : lerpDouble(middlePosition, trailingPosition, value - index)!,
    };
```

I believe the logic is identical, based on a local test run.

![image](https://github.com/flutter/flutter/assets/10457200/eecc1882-ccdc-498b-874b-a8eb8767d6a3)

Reply:
This is better, thanks!